lib/daru/io/io.rb

```diff
@@ -220,6 +232,41 @@ def from_csv_hash(path, opts)
 
         headers.each_with_index.map { |h, i| [h, csv_as_arrays[i]] }.to_h
       end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+      def parse_html_table(table)
+        data = table.search('tr').map { |row| row.search('td').map { |val| val.text.strip } }
+        size = data.map(&:count).max
+        data.keep_if { |x| x.count == size }
+        headers = table.search('tr').map { |row| row.search('th').map { |val| val.text.strip } }
+        return if headers[0].nil? || headers.map(&:count).max < size
+        headers.delete_at(0) while headers[0].nil? || headers[0].count < size
+        order = headers.delete_at(0)
+        ((order.delete_at 0) while order.count != size) if order.count > size
+        index = headers.flatten==[] ? nil : headers.flatten
+        return unless (index.nil? || index.count == data.count) && !order.nil? && order.count>0
+        {data: data.reject(&:empty?).reject(&:nil?), index: index, order: order}
+      end
+      # rubocop:enable Metrics/AbcSize, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+
+      def search_for_match(table, match=nil)
+        match.nil? ? true : (table.to_s.include? match)
+      end
+
+      def choose_value(scraped_val={}, user_val=nil)
+        unless user_val.nil?
+          user_val.each do |key,val|
+            scraped_val[key] = val
+          end
+        end
+        scraped_val
+      end
```

Comment:
What it does and why? Neither of method or parameter names are informative, and it is hard to understand the reasons.